common/src/verify.rs

```diff
@@ -75,27 +83,1425 @@ pub fn verify_finalization_proof(
+                },
+            ) => {
+                verify_header_to_header(&self.header, b)?;
+                // Verify block body
+                if *transaction_merkle_root != b.tx_merkle_root {
+                    return Err(Error::InvalidArgument(format!(
+                        "invalid transaction merkle root hash: expected {}, got {}",
+                        transaction_merkle_root, b.tx_merkle_root
+                    )));
+                }
+                // Verify commit hash
+                if self.commit_hash != b.commit_hash {
+                    return Err(Error::InvalidArgument(format!(
+                        "invalid block commit hash: expected {}, got {}",
+                        self.commit_hash, b.commit_hash
+                    )));
+                }
+                self.header = b.clone();
+                self.phase = Phase::Block;
+                self.commit_hash = Hash256::hash(format!("{}", b.height + 1));
+            }
+            (
+                Commit::Block(b),
+                Phase::ExtraAgendaTransaction {
+                    last_extra_agenda_timestamp,
+                    transaction_merkle_root,
+                },
+            ) => {
+                verify_header_to_header(&self.header, b)?;
+                // Check if the block contains all the extra-agenda transactions.
+                if b.timestamp < *last_extra_agenda_timestamp {
+                    return Err(Error::InvalidArgument(format!(
+                        "invalid block timestamp: expected larger than the last extra-agenda transaction timestamp {}, got {}",
+                        last_extra_agenda_timestamp, b.timestamp
+                    )));
+                }
+                // Verify block body
+                if *transaction_merkle_root != b.tx_merkle_root {
+                    return Err(Error::InvalidArgument(format!(
+                        "invalid transaction merkle root hash: expected {}, got {}",
+                        transaction_merkle_root, b.tx_merkle_root
+                    )));
+                }
+                // Verify commit hash
+                if self.commit_hash != b.commit_hash {
+                    return Err(Error::InvalidArgument(format!(
+                        "invalid block commit hash: expected {}, got {}",
+                        self.commit_hash, b.commit_hash
+                    )));
+                }
+                self.header = b.clone();
+                self.phase = Phase::Block;
+                self.commit_hash = Hash256::hash(format!("{}", b.height + 1));
+            }
+            (Commit::Block(_), Phase::Block) => {
+                return Err(Error::InvalidArgument(
+                    "invalid block commit: block commit already exists".to_string(),
+                ))
+            }
+            (Commit::Transaction(t), Phase::Block) => {
+                verify_validator(&self.header.validator_set, &t.author)?;
```

Comment:
This is not necessary. Transaction author may not be a member of the validator set.

Reply:
fixed.